lock-api/src/main/java/com/palantir/lock/watch/LockWatchEventCache.java

```diff
@@ -21,6 +21,11 @@
 import java.util.Set;
 
 public interface LockWatchEventCache {
+
+    default boolean isEnabled() {
+        return true;
+    }
```

Comment:
This feels wrong - this should call down to `LockWatchEventCacheImpl` or `NoOpLockWatchEventCache`.

Reply:
Oh I see - you went with an approach entirely through the proxy. I'm not sure I like that really - surely the logic is 100x cleaner just by doing what you originally said, i.e. having `LWECI` return true and `NOLWEC` return false?

Reply:
Discussed offline